src/coreclr/jit/assertionprop.cpp

```diff
@@ -168,6 +168,17 @@ bool IntegralRange::Contains(int64_t value) const
             }
             break;
 
+        case GT_IND:
+        {
+            if (node->gtGetOp1()->OperGet() == GT_ADD
+                && node->gtGetOp1()->gtGetOp1()->OperGet() == GT_LCL_VAR
+                && node->gtGetOp1()->gtGetOp1()->AsLclVar()->IsNeverNegative(compiler))
```

Comment:
Don't we also need to check that `op2` is also never negative? 

Or, if it can be negative, that it's most negative possible value won't make the sum negative?

Reply:
Yes, you're right and this is fragile and most likely won't work. I will redo this. Op2 should contain exactly the const offset of span.Length but adding another check might be too costly to cover the gains?

Is there a faster way to check if "this tree is Span.Length"? We have this information on gtFieldAddr but then it gets transformed to indir -> add (span, offset) and so information is lost. I may be wrong here, I will look more.

I looked at spmi diffs and they aren't huge. Around 700+800 bytes combined. FullOpt is around 450 bytes.

Not sure if some of those are actually bugs.